crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -216,57 +229,61 @@ pub(crate) enum ConnectionConnect {
 
 /// Resolves to a new Connection
 #[must_use = "futures do nothing unless polled"]
-pub struct StandardConnectionFuture(ConnectionConnect);
+pub struct TokioConnectionFuture {
+    connect: ConnectionConnect,
+    spawner: Handle,
+}
 
-impl Future for StandardConnectionFuture {
-    type Output = Result<(Connection, ConnectionBackground), ProtoError>;
+impl Future for TokioConnectionFuture {
+    type Output = Result<TokioConnection, ProtoError>;
 
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
-        let connection = match &mut self.0 {
+        let (connection, bg) = match &mut self.connect {
             ConnectionConnect::Udp(ref mut conn) => {
                 let (conn, bg) = ready!(conn.poll_unpin(cx))?;
-                let conn = Connection(ConnectionConnected::Udp(conn));
+                let conn = TokioConnection(ConnectionConnected::Udp(conn));
                 let bg = ConnectionBackground(ConnectionBackgroundInner::Udp(bg));
                 (conn, bg)
             }
             ConnectionConnect::Tcp(ref mut conn) => {
                 let (conn, bg) = ready!(conn.poll_unpin(cx))?;
-                let conn = Connection(ConnectionConnected::Tcp(conn));
+                let conn = TokioConnection(ConnectionConnected::Tcp(conn));
                 let bg = ConnectionBackground(ConnectionBackgroundInner::Tcp(bg));
                 (conn, bg)
             }
             #[cfg(feature = "dns-over-tls")]
             ConnectionConnect::Tls(ref mut conn) => {
                 let (conn, bg) = ready!(conn.poll_unpin(cx))?;
-                let conn = Connection(ConnectionConnected::Tls(conn));
+                let conn = TokioConnection(ConnectionConnected::Tls(conn));
                 let bg = ConnectionBackground(ConnectionBackgroundInner::Tls(bg));
                 (conn, bg)
             }
             #[cfg(feature = "dns-over-https")]
             ConnectionConnect::Https(ref mut conn) => {
                 let (conn, bg) = ready!(conn.poll_unpin(cx))?;
-                let conn = Connection(ConnectionConnected::Https(conn));
+                let conn = TokioConnection(ConnectionConnected::Https(conn));
                 let bg = ConnectionBackground(ConnectionBackgroundInner::Https(bg));
                 (conn, bg)
             }
             #[cfg(feature = "mdns")]
             ConnectionConnect::Mdns(ref mut conn) => {
                 let (conn, bg) = ready!(conn.poll_unpin(cx))?;
-                let conn = Connection(ConnectionConnected::Mdns(conn));
+                let conn = TokioConnection(ConnectionConnected::Mdns(conn));
                 let bg = ConnectionBackground(ConnectionBackgroundInner::Mdns(bg));
                 (conn, bg)
             }
         };
 
+        spawn_bg(&self.spawner, bg);
```

Comment:
@hawkw, this is now the current spawn in TokioConnectionProvider. It happens only after the Connect future runs. I think this meets all the previous goals.